contracts/consensus-gateway/ConsensusCogateway.sol

```diff
@@ -94,6 +100,96 @@ contract ConsensusCogateway is MasterCopyNonUpgradable, MessageBus, ConsensusGat
         );
     }
 
+    /**
+     * @notice This method will be called by anyone to verify merkle proof of
+     *          consensus gateway contract address.
+     *
+     *  @param _blockNumber Block number at which consensus gateway is to be proven.
+     *  @param _rlpAccount RLP encoded account node object.
+     *  @param _rlpParentNodes RLP encoded value of account proof node array.
+     */
+    function proveConsensusGateway(
+        uint256 _blockNumber,
+        bytes calldata _rlpAccount,
+        bytes calldata _rlpParentNodes
+    )
+        external
+    {
+        MessageInbox.proveStorageAccount(
+            _blockNumber,
+            _rlpAccount,
+            _rlpParentNodes
+        );
+    }
+
+    /**
+     * @notice Confirms the initiation of opening a kernel at auxiliary chain.
+     *
+     * @dev Function requires:
+     *          - Sender address must not be 0.
+     *          - Kernel hash must not be 0.
+     *          - Difference between kernelheight and current metablock height
+     *            must be 1.
+     *
+     * @param _kernelHeight Height of the kernel.
+     * @param _kernelHash Hash of the kernel.
+     * @param _feeGasPrice Gas price which the sender is willing to pay.
+     * @param _feeGasLimit Gas limit which the sender is willing to pay.
+     * @param _sender Sender address.
+     * @param _blockNumber Block number at which proof is valid.
+     * @param _rlpParentNodes RLP encoded parent node data to prove message
+     *                        exists in outbox of ConsensusGateway.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function confirmOpenKernel(
+        uint256 _kernelHeight,
+        bytes32 _kernelHash,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit,
+        address _sender,
+        uint256 _blockNumber,
+        bytes calldata _rlpParentNodes
+    )
+        external
+        returns (bytes32 messageHash_)
+    {
+        require(
+            _sender != address(0),
+            "Sender address must not be 0."
```

Comment:
Error message can be simplified: 'Sender address is 0.'

Similar comment for kernel hash error message